### DISCUSSION TITLE
Text being incorrectly parsed in table

### DISCUSSION THREAD

**Discussion by enrac5** [jsvine/pdfplumber#1149] [Ask for help with specific PDFs]
I have a very odd issue with the attached file:
[test.pdf](https://github.com/user-attachments/files/15688683/test.pdf)

Basically, the text in the second column is being read incorrectly and I'm not sure why. This is basically what I'm doing:

```
import pdfplumber

pdf_path = '/[path to attached]/test.pdf'
pdf = pdfplumber.open(pdf_path, repair=True)
for page in pdf.pages:
    tables = page.extract_tables()
    for table in tables:
        for row in table:
            if row[1] is not None:
                print(row[1])
```

The output should be `13+13` but I get `113+13`.  This is just a small part of a 78 page document (a PDF printed from MS-Word). (Puts on Leia's clothes, "Help me @jsvine, you're my only hope")

**Comment by jsvine** [comment]
Hi @enrac5, it looks like the PDF has two `1`s written on top of one another:

```python
print(page..extract_text(layout=True))
```

Produces:

```
...
1  113 +13 MT #1 FADES IN OVER  1 52+11  58+10  5+15 LUCY TO KOS
...
```

If you use `page.dedupe_chars()`, this seems to fix it:

```python
print(page.dedupe_chars().extract_text(layout=True))
``` 

```
...
1  13 +13 MT #1 FADES IN OVER   1 52+11  58+10  5+15 LUCY TO KOS
...
```

**Reply by enrac5** [reply]
Hi @jsvine thanks for this, though I have a dumb question. Does the `dedupe_chars()` call remove duplicate letters in a string?

**Reply by jsvine** [reply]
`dedupe_chars()` removes characters that are *on top of* the same character (rather than merely in the same string). See here for a slightly more detailed explanation: https://github.com/jsvine/pdfplumber?tab=readme-ov-file#extracting-text

**Reply by enrac5** [reply]
Hmm, does it not work with tables? When I do this:

```
import pdfplumber

pdf_path = '/path_to/test.pdf'
pdf = pdfplumber.open(pdf_path, repair=True)
for page in pdf.pages:
    tables = page.dedupe_chars().extract_tables()
    for table in tables:
        for row in table:
            if row[1] is not None:
                print(row[1])
```

I get the same wrong result.

**Reply by enrac5** [reply]
Also, when I copy and run your line:

```
print(page.dedupe_chars().extract_text(layout=True))
```

I still get the wrong result. I'm on version `0.10.3`, not sure if that changes anything.

**Reply by jsvine** [reply]
Hmmm. Seems to work fine for me on `0.10.3` and on `0.11.1`.

**Reply by enrac5** [reply]
Which Python version are you on? I'm on MacOS, testing (on a Jupyter Notebook) with Python 3.11.7

**Reply by enrac5** [reply]
When I spin up the Python prompt and enter:

```
>>> import pdfplumber
>>> pdf = pdfplumber.open('/path_to/test.pdf', repair=True)
>>> print(pdf.pages[0].dedupe_chars().extract_text(layout=True))
```

I get:

```
     Shot Footage Shot Description/Dialogue Title Start End Dur. Subtitle/Spotting  
                                                                                    
                                                                                    
     START MEASURING 0+00 AT PICTURE                                                
                                        LABORATORY NOTES                            
     START.                                                                         
                                        INSPECT SUBJECT’S ASS                       
                                                                                    
                                                                                    
      1  113+13 MT #1 FADES IN OVER   1 52+11  58+10  5+15 LUCY TO KOS              
                BLACK:                                     My ass has awakened.     
                                                                                    
                DARK ASS                                                            
                (animated logo)
```

**Reply by enrac5** [reply]
Weirdly, when I don't specify `repair=True` in the `open(...)` call, I get:

```
     Shot Footage Shot Description/Dialogue Title Start End Dur. Subtitle/Spotting  
                                                                                    
                                                                                    
     START MEASURING 0+00 AT PICTURE                                                
                                        LABORATORY NOTES                            
     START.                                                                         
                                        INSPECT SUBJECT’S ASS                       
                                                                                    
                                                                                    
      1  1 13+13 MT #1 FADES IN OVER  1 52+11  58+10  5+15 LUCY TO KOS              
                BLACK:                                     My ass has awakened.     
                                                                                    
                DARK ASS                                                            
                (animated logo) 
```

The entry that was `113+13` is now `1 13+13`, which is still wrong, but different.

**Reply by jsvine** [reply]
Strange. I run the exact same code as you but get the correct output. 

<img width="678" alt="Screenshot 2024-06-14 at 6 45 01 PM" src="https://github.com/jsvine/pdfplumber/assets/534702/250654ff-b470-4312-a8cd-7bf643a0e584">

Ran that on Python 3.10.4 on MacOS, though neither should not have any effect on the processing. Try upgrading `pdfplumber` and running `print(pdfplumber.__version__)` to confirm?

**Reply by enrac5** [reply]
Which version of MacOS are you on?

**Reply by jsvine** [reply]
For an operation like this one, the operating system should not have any effect on the results. (Both theoretically, and from practical experience; I haven't encountered any PDF where an operation such as `dedupe_chars` performs differently on different OSes.) Different versions of Ghostscript (used for `repair=True`), however, may have an effect, so I'd focus on the output of *without* `repair=True`.

Is it possible you're processing a slightly different version of the PDF than the one shared in this issue?

**Reply by enrac5** [reply]
I'm using the same PDF, though this is a stripped down example of another one I cannot share.

**Reply by enrac5** [reply]
On my WSL instance, this works amazingly, however on my MacOS install, it doesn't, that's the flummoxing part.
